packages/lib/src/io-command.ts

```diff
@@ -653,6 +659,27 @@ async function stringTranslateToId<L>(this: APICommand & { readonly primaryKeyNa
 		` ${this.primaryKeyName} in ${JSON.stringify(matchingItem)}`)
 }
 
+async function stringTranslateToId2<ID, L2>(this: APICommand & { readonly primaryKeyName2: string; sort2(list: L2[]): L2[] },
+		id: ID,
+		idOrIndex2: string,
+		listFunction: NestedListCallback<ID, L2>): Promise<string> {
+
+	let pk2 = idOrIndex2
+	if (isIndexArgument(idOrIndex2)) {
+		const index = Number.parseInt(idOrIndex2)
+		const items = this.sort2(await listFunction(id))
+		const matchingItem: L2 = items[index - 1]
+		if (!(this.primaryKeyName2 in matchingItem)) {
+			throw Error(`did not find key ${this.primaryKeyName2} in data ${JSON.stringify(items)}`)
+		}
+
+		// @ts-ignore
```

Comment:
If we cannot find ways around these, I suggest we start commenting which error(s) we are suppressing [as recommended by Microsoft](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments). This will make it easier to debug later if something further unexpected starts happening on these lines.

This comment applies for all 7 usages of `@ts-ignore` in this diff.

Reply:
Fixed